clients/src/main/java/org/apache/kafka/clients/consumer/internals/MemberState.java

```diff
@@ -24,21 +24,34 @@
 public enum MemberState {
 
     /**
-     * Member has not joined a consumer group yet, or has been fenced and needs to re-join.
+     * Member is not part of the group. This could be the case when it has never joined (no call
+     * has been made to the subscribe API), or when the member intentionally leaves the group
+     * after a call to the unsubscribe API.
      */
-    UNJOINED,
+    NOT_IN_GROUP,
```

Comment:
You're right, this is the case where a consumer, with a groupId, is not part of the group (either it hasn't called subscribed, or it called unsubscribe)

Reply:
`Unsubscribed` describes the state better to me too. Renamed it and added comments explaining better how the member gets there and what it can do while in this state.